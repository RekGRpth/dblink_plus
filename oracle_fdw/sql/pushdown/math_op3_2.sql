\c contrib_regression_utf8

SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';

-- '*' operator
-- 141 : int4mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_integer_1 = 1;
-- 152 : int2mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_smallint_1 = 1;
-- 170 : int24mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_integer_1 = 1;
-- 171 : int42mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_smallint_1 = 1;
-- 202 : float4mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 * binary_float_real_1 = 1;
-- 216 : float8mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 * binary_double_precision_1 = 1;
-- 279 : float48mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 * binary_double_precision_1 = 1;
-- 283 : float84mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 * binary_float_real_1 = 1;
-- 465 : int8mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_bigint_1 = 1;
-- 839 : int82mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_smallint_1 = 1;
-- 943 : int28mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_bigint_1 = 1;
-- 1276 : int84mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_integer_1 = 1;
-- 1280 : int48mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_bigint_1 = 1;
-- 1726 : numeric_mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 * number_numeric3_1 = 1;

-- 141 : int4mul
SELECT id FROM all_type WHERE number_integer_1 * number_integer_1 = 1;
-- 152 : int2mul
SELECT id FROM all_type WHERE number_smallint_1 * number_smallint_1 = 1;
-- 170 : int24mul
SELECT id FROM all_type WHERE number_smallint_1 * number_integer_1 = 1;
-- 171 : int42mul
SELECT id FROM all_type WHERE number_integer_1 * number_smallint_1 = 1;
-- 202 : float4mul
SELECT id FROM all_type WHERE binary_float_real_1 * binary_float_real_1 = 1;
-- 216 : float8mul
SELECT id FROM all_type WHERE binary_double_precision_1 * binary_double_precision_1 = 1;
-- 279 : float48mul
SELECT id FROM all_type WHERE binary_float_real_1 * binary_double_precision_1 = 1;
-- 283 : float84mul
SELECT id FROM all_type WHERE binary_double_precision_1 * binary_float_real_1 = 1;
-- 465 : int8mul
SELECT id FROM all_type WHERE number_bigint_1 * number_bigint_1 = 1;
-- 839 : int82mul
SELECT id FROM all_type WHERE number_bigint_1 * number_smallint_1 = 1;
-- 943 : int28mul
SELECT id FROM all_type WHERE number_smallint_1 * number_bigint_1 = 1;
-- 1276 : int84mul
SELECT id FROM all_type WHERE number_bigint_1 * number_integer_1 = 1;
-- 1280 : int48mul
SELECT id FROM all_type WHERE number_integer_1 * number_bigint_1 = 1;
-- 1726 : numeric_mul
SELECT id FROM all_type WHERE number_numeric3_1 * number_numeric3_1 = 1;
