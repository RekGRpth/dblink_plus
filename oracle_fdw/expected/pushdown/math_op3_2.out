\c contrib_regression_utf8
SET DateStyle = 'ISO,YMD';
SET IntervalStyle = 'sql_standard';
SET TIMEZONE = 'Asia/Tokyo';
-- '*' operator
-- 141 : int4mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_integer_1 = 1;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" * "number_integer_1") = 1)
(2 rows)

-- 152 : int2mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_smallint_1 = 1;
                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                          
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" * "number_smallint_1") = 1)
(2 rows)

-- 170 : int24mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_integer_1 = 1;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                          
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" * "number_integer_1") = 1)
(2 rows)

-- 171 : int42mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_smallint_1 = 1;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                          
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" * "number_smallint_1") = 1)
(2 rows)

-- 202 : float4mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 * binary_float_real_1 = 1;
                                                                                                                                                                                                                                            QUERY PLAN                                                                                                                                                                                                                                            
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" * "binary_float_real_1") = 1)
(2 rows)

-- 216 : float8mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 * binary_double_precision_1 = 1;
                                                                                                                                                                                                                                                  QUERY PLAN                                                                                                                                                                                                                                                  
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" * "binary_double_precision_1") = 1)
(2 rows)

-- 279 : float48mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_float_real_1 * binary_double_precision_1 = 1;
                                                                                                                                                                                                                                               QUERY PLAN                                                                                                                                                                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_float_real_1" * "binary_double_precision_1") = 1)
(2 rows)

-- 283 : float84mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE binary_double_precision_1 * binary_float_real_1 = 1;
                                                                                                                                                                                                                                               QUERY PLAN                                                                                                                                                                                                                                               
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("binary_double_precision_1" * "binary_float_real_1") = 1)
(2 rows)

-- 465 : int8mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_bigint_1 = 1;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                        
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" * "number_bigint_1") = 1)
(2 rows)

-- 839 : int82mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_smallint_1 = 1;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" * "number_smallint_1") = 1)
(2 rows)

-- 943 : int28mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_smallint_1 * number_bigint_1 = 1;
                                                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_smallint_1" * "number_bigint_1") = 1)
(2 rows)

-- 1276 : int84mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_bigint_1 * number_integer_1 = 1;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                         
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_bigint_1" * "number_integer_1") = 1)
(2 rows)

-- 1280 : int48mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_integer_1 * number_bigint_1 = 1;
                                                                                                                                                                                                                                        QUERY PLAN                                                                                                                                                                                                                                         
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_integer_1" * "number_bigint_1") = 1)
(2 rows)

-- 1726 : numeric_mul
EXPLAIN (COSTS FALSE) SELECT id FROM all_type WHERE number_numeric3_1 * number_numeric3_1 = 1;
                                                                                                                                                                                                                                          QUERY PLAN                                                                                                                                                                                                                                          
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Foreign Scan on all_type
   Remote SQL: SELECT "id", NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL FROM "oracle_fdw"."all_type" WHERE (("number_numeric3_1" * "number_numeric3_1") = 1)
(2 rows)

-- 141 : int4mul
SELECT id FROM all_type WHERE number_integer_1 * number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 152 : int2mul
SELECT id FROM all_type WHERE number_smallint_1 * number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 170 : int24mul
SELECT id FROM all_type WHERE number_smallint_1 * number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 171 : int42mul
SELECT id FROM all_type WHERE number_integer_1 * number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 202 : float4mul
SELECT id FROM all_type WHERE binary_float_real_1 * binary_float_real_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 216 : float8mul
SELECT id FROM all_type WHERE binary_double_precision_1 * binary_double_precision_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 279 : float48mul
SELECT id FROM all_type WHERE binary_float_real_1 * binary_double_precision_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 283 : float84mul
SELECT id FROM all_type WHERE binary_double_precision_1 * binary_float_real_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 465 : int8mul
SELECT id FROM all_type WHERE number_bigint_1 * number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 839 : int82mul
SELECT id FROM all_type WHERE number_bigint_1 * number_smallint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 943 : int28mul
SELECT id FROM all_type WHERE number_smallint_1 * number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1276 : int84mul
SELECT id FROM all_type WHERE number_bigint_1 * number_integer_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1280 : int48mul
SELECT id FROM all_type WHERE number_integer_1 * number_bigint_1 = 1;
 id 
----
  5
  7
(2 rows)

-- 1726 : numeric_mul
SELECT id FROM all_type WHERE number_numeric3_1 * number_numeric3_1 = 1;
 id 
----
  5
  7
(2 rows)

